packages/alfa-rules/src/sia-r7/rule.ts

```diff
@@ -3,22 +3,26 @@ import { Attribute, Element, Namespace } from "@siteimprove/alfa-dom";
 import { Language } from "@siteimprove/alfa-iana";
 import { Iterable } from "@siteimprove/alfa-iterable";
 import { Predicate } from "@siteimprove/alfa-predicate";
+import { Refinement } from "@siteimprove/alfa-refinement";
 import { Err, Ok } from "@siteimprove/alfa-result";
 import { Criterion, Technique } from "@siteimprove/alfa-wcag";
 import { Page } from "@siteimprove/alfa-web";
 
 import { expectation } from "../common/expectation";
+import { hasDescendant, isIgnored } from "../common/predicate";
 
 import { hasAttribute } from "../common/predicate/has-attribute";
+import { isVisible } from "../common/predicate/is-visible";
 
 const { isElement, hasName, hasNamespace } = Element;
 const { isEmpty } = Iterable;
-const { and, not } = Predicate;
+const { and, not, or } = Predicate;
+const { and: andRefinement } = Refinement;
```

Comment:
```suggestion
const { not, or } = Predicate;
const { and } = Refinement;
```

When passed "normal" predicates, `Refinement.and()` behaves like `Predicate.and()`.